exporter/awsemfexporter/metric_translator.go

```diff
@@ -385,10 +424,8 @@ func dimensionRollup(dimensionRollupOption string, originalDimensionSlice []stri
 	}
 	if dimensionRollupOption == ZeroAndSingleDimensionRollup || dimensionRollupOption == SingleDimensionRollupOnly {
 		//"One" dimension rollup
-		if len(originalDimensionSlice) > 1 {
-			for _, dimensionKey := range originalDimensionSlice {
-				rollupDimensionArray = append(rollupDimensionArray, append(dimensionZero, dimensionKey))
-			}
+		for _, dimensionKey := range originalDimensionSlice {
+			rollupDimensionArray = append(rollupDimensionArray, append(dimensionZero, dimensionKey))
```

Comment:
we only do rollup when orig dimension len >=2, it was added in this commit to fix a bug. Any reason to remove it now?

https://github.com/open-telemetry/opentelemetry-collector-contrib/commit/d4fe7c3eec1e76328ea32066d91f934a0f02866e#diff-af3bf0e3a5a889da31842d592ffeecc4ef5d3e5b058af7ca500ed44e321bca0cR437

Reply:
Yup, this change did not work well with metric declarations since this assumes that the exported dimensions are only the labels. I implemented the dimension de-duping logic that handles this case and also for the metric declaration case: https://github.com/hdj630/opentelemetry-collector-contrib/blob/5d8cbbd67d2c3d6fa3c83abddfc87fa676f2df45/exporter/awsemfexporter/metric_translator.go#L292

Reply:
I see. Thx.